Data Page Refresh + Plugin x Project KPI (#286)
meltano/handbook: src/_data-team/kpis.md

Certain KPIs are considered Health Metrics, which means we actively monitor them and alarm on them and take action when and if they fall out of expected range. Otherwise we may or may not have goals or initiatives associated with their outputs.

> ## Projects x Plugins Score (PPS)

Reviewer: ```suggestion
## Projects x Plugins Score (PXP)
```

PXP just sounds cooler 😎

> Meltano's primary strategic goal is to achieve our vision towards product-market fit for Meltano as the DataOps platform infrastructure.

Reviewer: ```suggestion
Meltano's primary strategic goal is to achieve product-market fit for Meltano as the DataOps Platform Infrastructure.
```

> Our primary metric for measuring progress towards this goal is the number of projects times the amount of plugins categories that they use.

Reviewer: ```suggestion
Our primary metric for measuring progress towards this goal is the number of projects times the amount of plugins categories used in each project.
```

The metric is aggregated daily over a 14 day rolling window excluding executions that are within 7 days of their first event.
The intention is to reduce the noise created by first time users exploring so the metric more closely tracks long term users.
A 14 day window was used because we want to monitor on a shorter timeline, ideally 1 week, to understand how our actions are contributing to the metric but
7 days ends up being too short potentially due to projects that run pipelines less frequently.
Only execution events are considered for this metric meaning invoke/elt/run/test (ui doesnt execute a plugin so it gets naturally excluded even though its an execution event).
If the execution project ID source is "random" then its not considered because we can't confidently differentiate that project from another we've already counted so we risk over counting.
PPS is used to monitor both an increase in overall projects using Meltano as well as the variety of plugins that those projects use.

> For example, 10 projects using only Singer or 2 using Singer/dbt/GE/Airflow/Superset would both count as 10 "points" toward the PPS.

Reviewer: ```suggestion
For example, 10 projects using only Singer (1 plugin category) or 2 projects using Singer + dbt + GE + Airflow + Superset (5 categories) would both count as 10 "points" toward the PXP.
```

Let's move this and the final paragraph under the first paragraph so the flow is:

* Intro
* Example of how calculated - this plus the intro is the "bottom line up front"
* Why it's useful
* Details about the calculation

In fact, it might be useful to have H3 or H4 header for "Implementation Details" on the metric. Most people won't need to understand the details, but it's good to have them there and to be transparent about it.

Author: @tayloramurphy I accepted your suggestion, I'll shift the ordering around like you suggested.

Certain KPIs are considered Health Metrics, which means we actively monitor them and alarm on them and take action when and if they fall out of expected range. Otherwise we may or may not have goals or initiatives associated with their outputs.

## Projects x Plugins Score (PXP)

Meltano's primary strategic goal is to achieve product-market fit for Meltano as the DataOps Platform Infrastructure.
Our primary metric for measuring progress towards this goal is the number of projects times the amount of plugins categories used in each project.

For example, 10 projects using only Singer (1 plugin category) or 2 projects using Singer + dbt + GE + Airflow + Superset (5 categories) would both count as 10 "points" toward the PXP.

This metric is used to monitor both the active user base in general as well as the variety of plugins that those projects use.
This metric encourages us to both grow the existing user base while also encouraging users to increase the variety of plugins they use.
This means the existing projects implemented pre-2.0 only using EL still positively contribute to this metric.

#### Implementation Details

> The metric is aggregated daily over a 14 day rolling window excluding executions that are within 7 days of their first event.

Reviewer 2: This also means that projects that are abandoned within 7 days are not included at all, right? Can we make that explicit?

Author: @DouweM yep good call out, I can make that more explicit.

The intention is to reduce the noise created by first time users exploring so the metric more closely tracks long term users.

> A 14 day window was used because we want to monitor on a shorter timeline, ideally 1 week, to understand how our actions are contributing to the metric but

Reviewer 2: Since we filter out events within 7 days of project creation, data on new projects would be lagging by a week, but older projects increasing their plugin usage would show up immediately, right? Since there's a delay there anyway, how does the 14d window help with monitoring on a shorter timeline in a way that 28d wouldn't?

Author: > Since we filter out events within 7 days of project creation, data on new projects would be lagging by a week, but older projects increasing their plugin usage would show up immediately, right?

@DouweM yes projects that are older than 7 days will have their new plugins counted immediately.

> Since there's a delay there anyway, how does the 14d window help with monitoring on a shorter timeline in a way that 28d wouldn't?

My thought was that its more sensitive, if someone tries out a plugin once, then stops using it we only get to account for it for 14 days before it gets dropped from our count. If we use a 28 day window then 1 test execution would give us a full month of credit for it. I think during our 1-1 we were talking about how 7 days would be ideal but then I adjusted to 14 days after noticing it didnt seem to be capturing as much as expected.

Does that make sense? I proposed this documentations along side the code changes so we can have this discussion, I'm totally open to adjusting.

Reviewer 2: @pnadolny13 Ah right, 2 weeks makes sense then.

Author: @DouweM yes totally fair point - from our chat on the side theres pros and cons of both approaches. It sounds like we're leaning towards 28d instead.

@tayloramurphy do you have any strong opinions on 28d vs 14d vs 7d?

Reviewer: @pnadolny13 for the top-level number I can see the argument for 28d. Personally I'd like to see the 7d and 14d variants to see trends and how it's affected. 

My vote would be for 28d for the main number but then a deeper view that has all 3 that we wouldn't need to share externally.
